runtime/doc/api.txt

```diff
@@ -923,6 +923,26 @@ nvim_get_current_win()                                *nvim_get_current_win()*
     Return: ~
         Window handle
 
+nvim_get_hl({ns_id}, {*val})                                   *nvim_get_hl()*
+    Gets highlight groups
+
+    Note:
+        Unlike the `:highlight` command which lists all the current highlight
+        groups that have attributes set in human readable format, this
+        function returns a table of all current highlight groups in a machine
+        readable format (i.e. can be directly fed to nvim_set_hl)
+
+    Parameters: ~
+      • {ns_id}  Get highlight groups using namespace ns_id
+                 |nvim_get_namespaces()|. Use 0 to get global highlight groups
+                 |:highlight|.
+      • {val}    Options dict:
+                 • name: (string) Get a highlight definition by name.
+                 • id: (integer) Get a highlight definition by id.
+                 • hex: (boolean, default false) shows colors in RGB hex
+                   representation instead of decimal.
```

Comment:
I would be good to clarify that `hex representation` means a string. I would expect consumers to mostly use decimal and use `string.format('%x')` when they want to stringify it.

Reply:
I don't think so, the initial goal of this function is to manipulate colorscheme, I think `#1e1e2e" is better default than `113392728`

Reply:
How do you manipulate a colorscheme with the string form? I've only ever manipulated using the integer form (as above).

Reply:
Sorry, why does returning an integer break this? The highlights should be the same. `nvim_set_hl` accepts integers for `fg` and bg`.

Reply:
Ah I meant the `(r, g, b)` tuple you mentioned

Reply:
Now that we have `bit` guaranteed in the lua stdlib, we can add further (pure-lua) helper functions to split a 24-bit integer into three 8-bit integers and vice versa. It doesn't need to be present in the language-independent API-interface.

Reply:
> Should I remove the hex option from this PR?

That would be my vote unless there's a use case I'm not aware of.